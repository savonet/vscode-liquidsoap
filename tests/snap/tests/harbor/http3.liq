def f()
  content_type = http.headers.content_type([
    ("Content-Type", "text/html; charset=utf-8")
  ])
  test.equals(content_type, { mime="text/html", args=[("charset", "utf-8")] })

  content_type = http.headers.content_type([
    ("coNtent-tYpe", "multipart/form-data; boundary=something")
  ])
  test.equals(content_type, { mime="multipart/form-data", args=[("boundary", "something")] })

  extname = http.headers.extname([
    ("Content-Disposition", 'attachment; filename="filename.jpg"')
  ])
  test.equals(extname, ".jpg")

  extname = http.headers.extname([
    ("Content-type", "audio/mpeg"),
    ("Content-Disposition", 'attachment; filename="filename.jpg"')
  ])
  test.equals(extname, ".jpg")

  settings.http.mime.extnames := [("foo", ".bla")]

  extname = http.headers.extname([
    ("conTent-tyPe", "foo"),
  ])
  test.equals(extname, ".bla")

  test.pass()
end

test.check(f)
