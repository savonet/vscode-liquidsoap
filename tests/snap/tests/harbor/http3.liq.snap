>def f()
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^ source.liquidsoap
>  content_type = http.headers.content_type([
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>    ("Content-Type", "text/html; charset=utf-8")
#^^^^^ source.liquidsoap
#     ^ source.liquidsoap string.quoted.double.liquidsoap
#      ^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                  ^ source.liquidsoap string.quoted.double.liquidsoap
#                   ^^ source.liquidsoap
#                     ^ source.liquidsoap string.quoted.double.liquidsoap
#                      ^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                              ^ source.liquidsoap string.quoted.double.liquidsoap
#                                               ^^ source.liquidsoap
>  ])
#^^^^^ source.liquidsoap
>  test.equals(content_type, { mime="text/html", args=[("charset", "utf-8")] })
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                   ^ source.liquidsoap string.quoted.double.liquidsoap
#                                    ^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                             ^ source.liquidsoap string.quoted.double.liquidsoap
#                                              ^^^^^^^^^ source.liquidsoap
#                                                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                        ^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                                               ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                ^^ source.liquidsoap
#                                                                  ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                   ^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                        ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                         ^^^^^^ source.liquidsoap
>
>  content_type = http.headers.content_type([
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>    ("coNtent-tYpe", "multipart/form-data; boundary=something")
#^^^^^ source.liquidsoap
#     ^ source.liquidsoap string.quoted.double.liquidsoap
#      ^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                  ^ source.liquidsoap string.quoted.double.liquidsoap
#                   ^^ source.liquidsoap
#                     ^ source.liquidsoap string.quoted.double.liquidsoap
#                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                                             ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                              ^^ source.liquidsoap
>  ])
#^^^^^ source.liquidsoap
>  test.equals(content_type, { mime="multipart/form-data", args=[("boundary", "something")] })
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                   ^ source.liquidsoap string.quoted.double.liquidsoap
#                                    ^^^^^^^^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                        ^^^^^^^^^ source.liquidsoap
#                                                                 ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                  ^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                          ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                           ^^ source.liquidsoap
#                                                                             ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                              ^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                                                        ^^^^^^ source.liquidsoap
>
>  extname = http.headers.extname([
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>    ("Content-Disposition", 'attachment; filename="filename.jpg"')
#^^^^^ source.liquidsoap
#     ^ source.liquidsoap string.quoted.double.liquidsoap
#      ^^^^^^^^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                         ^ source.liquidsoap string.quoted.double.liquidsoap
#                          ^^ source.liquidsoap
#                            ^ source.liquidsoap string.quoted.single.liquidsoap
#                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap string.quoted.single.liquidsoap
#                                                                ^ source.liquidsoap string.quoted.single.liquidsoap
#                                                                 ^^ source.liquidsoap
>  ])
#^^^^^ source.liquidsoap
>  test.equals(extname, ".jpg")
#^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                        ^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                            ^ source.liquidsoap string.quoted.double.liquidsoap
#                             ^^ source.liquidsoap
>
>  extname = http.headers.extname([
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>    ("Content-type", "audio/mpeg"),
#^^^^^ source.liquidsoap
#     ^ source.liquidsoap string.quoted.double.liquidsoap
#      ^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                  ^ source.liquidsoap string.quoted.double.liquidsoap
#                   ^^ source.liquidsoap
#                     ^ source.liquidsoap string.quoted.double.liquidsoap
#                      ^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                ^ source.liquidsoap string.quoted.double.liquidsoap
#                                 ^^^ source.liquidsoap
>    ("Content-Disposition", 'attachment; filename="filename.jpg"')
#^^^^^ source.liquidsoap
#     ^ source.liquidsoap string.quoted.double.liquidsoap
#      ^^^^^^^^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                         ^ source.liquidsoap string.quoted.double.liquidsoap
#                          ^^ source.liquidsoap
#                            ^ source.liquidsoap string.quoted.single.liquidsoap
#                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap string.quoted.single.liquidsoap
#                                                                ^ source.liquidsoap string.quoted.single.liquidsoap
#                                                                 ^^ source.liquidsoap
>  ])
#^^^^^ source.liquidsoap
>  test.equals(extname, ".jpg")
#^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                        ^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                            ^ source.liquidsoap string.quoted.double.liquidsoap
#                             ^^ source.liquidsoap
>
>  settings.http.mime.extnames := [("foo", ".bla")]
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                   ^ source.liquidsoap string.quoted.double.liquidsoap
#                                    ^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                                        ^^ source.liquidsoap
#                                          ^ source.liquidsoap string.quoted.double.liquidsoap
#                                           ^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                               ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                ^^^ source.liquidsoap
>
>  extname = http.headers.extname([
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>    ("conTent-tyPe", "foo"),
#^^^^^ source.liquidsoap
#     ^ source.liquidsoap string.quoted.double.liquidsoap
#      ^^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                  ^ source.liquidsoap string.quoted.double.liquidsoap
#                   ^^ source.liquidsoap
#                     ^ source.liquidsoap string.quoted.double.liquidsoap
#                      ^^^ source.liquidsoap string.quoted.double.liquidsoap
#                         ^ source.liquidsoap string.quoted.double.liquidsoap
#                          ^^^ source.liquidsoap
>  ])
#^^^^^ source.liquidsoap
>  test.equals(extname, ".bla")
#^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                        ^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                            ^ source.liquidsoap string.quoted.double.liquidsoap
#                             ^^ source.liquidsoap
>
>  test.pass()
#^^^^^^^^^^^^^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
>test.check(f)
#^^^^^^^^^^^^^^ source.liquidsoap
>